modules/objdetect/src/qrcode.cpp

```diff
@@ -3615,18 +3615,17 @@ bool QRCodeDetector::decodeMulti(
             for_copy.push_back(straight_barcode[i]);
     }
     straight_barcode = for_copy;
-    vector<Mat> tmp_straight_qrcodes;
-    if (straight_qrcode.needed())
+    if (straight_qrcode.needed() &&
+       (straight_barcode.size() > 0))
     {
+        vector<Mat> tmp_straight_qrcodes(straight_barcode.size());
         for (size_t i = 0; i < straight_barcode.size(); i++)
         {
-            Mat tmp_straight_qrcode;
-            tmp_straight_qrcodes.push_back(tmp_straight_qrcode);
-            straight_barcode[i].convertTo(((OutputArray)tmp_straight_qrcodes[i]),
-                                             ((OutputArray)tmp_straight_qrcodes[i]).fixedType() ?
-                                             ((OutputArray)tmp_straight_qrcodes[i]).type() : CV_32FC2);
+            straight_barcode[i].convertTo(tmp_straight_qrcodes[i],
+                                          straight_qrcode.fixedType() ?
+                                          straight_qrcode.type() : CV_32FC2);
```

Comment:
> CV_32FC2

What is stored here?

Documentation says:

> The optional output vector of images containing rectified and binarized QR codes
> The optional output image containing rectified and binarized QR code

What is about simple tests for checking output parameter types?

Reply:
> binarized

https://en.wikipedia.org/wiki/Binary_image

Why do we need floating point type here? `8UC1` is enough.